libpod/container_internal.go

```diff
@@ -2738,3 +2741,327 @@ func (c *Container) update(resources *spec.LinuxResources, restartPolicy *string
 
 	return nil
 }
+
+func (c *Container) getCopyOfHealthCheckAndStartupHelathCheck() (*manifest.Schema2HealthConfig, *define.StartupHealthCheck) {
+	var healthCheck manifest.Schema2HealthConfig
+	if c.config.HealthCheckConfig != nil {
+		healthCheck = manifest.Schema2HealthConfig{
+			Test:          slices.Clone(c.config.HealthCheckConfig.Test),
+			Interval:      c.config.HealthCheckConfig.Interval,
+			Retries:       c.config.HealthCheckConfig.Retries,
+			StartInterval: c.config.HealthCheckConfig.StartInterval,
+			StartPeriod:   c.config.HealthCheckConfig.StartPeriod,
+			Timeout:       c.config.HealthCheckConfig.Timeout,
+		}
+	}
+
+	var startupHealthCheck define.StartupHealthCheck
+	if c.config.StartupHealthCheckConfig != nil {
+		startupHealthCheck = define.StartupHealthCheck{
```

Comment:
Same here, investigate JSONDeepCopy, I don't think this is performance critical at all.